test/extended/testdata/olm/image-catalogsource.yaml

```diff
@@ -0,0 +1,10 @@
+apiVersion: operators.coreos.com/v1alpha1
+kind: CatalogSource
+metadata:
+  name: test-operator
+  namespace: openshift-marketplace
+spec:
+  sourceType: grpc
+  image: quay.io/aosqe42/operator-registry:auto
```

Comment:
What publishes this image? Is it easily reproducible? Should it come from a build system?

Reply:
It was build by using the `operator-registry`. As follows:
```console
1, git clone https://github.com/operator-framework/operator-registry

2, mac:operator-registry jianzhang$ cd manifests/
mac:manifests jianzhang$ ls
etcd                   prometheus             strimzi-kafka-operator
mac:manifests jianzhang$ rm -rf prometheus strimzi-kafka-operator
mac:manifests jianzhang$ 
3, mac:manifests jianzhang$ tree
.
└── etcd
    ├── 0.6.1
    │   ├── etcdcluster.crd.yaml
    │   └── etcdoperator.clusterserviceversion.yaml
    ├── 0.9.2
    │   ├── etcdbackup.crd.yaml
    │   ├── etcdcluster.crd.yaml
    │   ├── etcdoperator.v0.9.0.yaml
    │   ├── etcdoperator.v0.9.2.clusterserviceversion.yaml
    │   └── etcdrestore.crd.yaml
    └── etcd.package.yaml

3 directories, 8 files

4, mac:operator-registry jianzhang$ docker build -f upstream-example.Dockerfile  -t quay.io/aosqe42/operator-registry:auto2 .
Sending build context to Docker daemon  81.88MB
```

Reply:
```console
mac:operator-registry jianzhang$ docker push quay.io/aosqe42/operator-registry:auto2 
```

Reply:
Here is the related manifest: https://github.com/jianzhangbjz/operator-registry/tree/pr-23186/manifests/etcd-test

Reply:
So I think relying on a locally built image for this test is somewhat problematic. Can we try to use a reproducible image based on a build environment? What happens when something about this image needs to/does change and we don't have automation in place to regenerate this image? Who owns this namespace and how can we update the images in it? Can we at the very least have some common credentials and a common build environment so that we can migrate this over time? At the very least some documentation so that someone can decipher how to do this two years from now if none of us are around anymore?

Reply:
Alternatively, could we just use a config map for this test for now? We have been thinking about how it would be possible to generate these images at test time, but right now the build environment doesn't have an obvious way to generate a container image that can be pushed to a registry while a test is actually running.